### PR TITLE
feat: use statePath for inferred @DataAction() type

### PR DESCRIPTION
**Pull request by bitflut**
This way we can better identify children reducer actions via type

### CODE REVIEW

**Comment by bitflut** [comment]
I'll fix the linting issues in a minute

**Comment by splincode** [comment]
@bitflut Good work! Could also add some new tests that solve the problem with child states and paths. I would like to see what problem we solved))

**Comment by splincode** [comment]
I mean add test and add NgxsLoggerPluginModule and add spyOn for logging

`deep-name.spec.ts`

```ts
@State({ 
  name: 'c'
})
class C {}

@State({ 
  name: 'c',
  children: [C]
})
class B {}

@State({ 
  name: 'a',
  children: [B]
})
class A {}

a.setState();
b.setState();
c.setState();
```

File: lib/internals/src/utils/action/action-name-creator.ts [hunk 1]
```diff
@@ -25,5 +25,5 @@ export function actionNameCreator(options: ActionNameCreatorOptions): string {
         }
     }
 
-    return `@${stateName}.${methodName}(${argsList})`;
+    return `@${statePath.split('.').join('/')}.${methodName}(${argsList})`;
```
Comment (splincode) [inline lib/internals/src/utils/action/action-name-creator.ts:28]: what do you think about 

```ts
statePath.replace(/\./g, '/')
```

instead 

```ts
statePath.split('.').join('/')
```

Comment (bitflut) [reply lib/internals/src/utils/action/action-name-creator.ts:28]: sure, if that is more performant 👍

Comment (splincode) [reply lib/internals/src/utils/action/action-name-creator.ts:28]: investigate permormance?

Comment (bitflut) [reply lib/internals/src/utils/action/action-name-creator.ts]: Seems like your intuition was right. Let's go with replace: 
https://stackoverflow.com/a/50463973/341837

**Comment by splincode** [comment]
please update changelog

**Pull request merged**